### PR TITLE
Adquirir tecnologia

### LINKED ISSUE #501: Adquirir tecnologia
## Feature Description

No detalhamento da tecnologia, ao usuário clicar no botão QUERO ADQUIRIR ESSA TECNOLOGIA:

Quero adquirir essa tecnologia

 Apresentar o seguinte modal:

form-aquisição

Os campos necessários para a aquisição são:
- Quantas unidades da tecnologia: 
- A tecnologia será adquirida para uso: (Privado, Empresa, Municipal, Estadual, Federal, Outro)
- Deseja financiamento para essa aquisição? (Sim, eu já tenho como financiar, Sim, mas não tenho como financiar, Não preciso de financiamento)
- Observações
- Finalizar com botão de ADQUIRIR TECNOLOGIA e CANCELAR
 
---------------

## Acceptance criteria

* 

- Somente usuários logados e com o cadastro de dados pessoais completos poderão adquirir a tecnologia
- O sistema precisa checar se o usuário completou os dados pessoais de cadastro, caso contrário, precisa desviar o fluxo para esse preenchimento.
- Uma notificação é feita ao pesquisador dono da tecnologia
- Uma negociação é registrada em Meus Pedidos na área do usuário
- Uma negociação é registrada em Pedidos na área do pesquisador

## Implementation Brief

*

### PR DESCRIPTION
**Pull request by luizeboli**
## Summary

Resolves #501

## Relevant technical choices

* API isn't done yet, we decided to integrate when it's ok.

## QA Steps

1. Access any technology details page
2. Click **Quero adquirir essa tecnologia** button

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code passes the linting.
- [x] My code has proper inline documentation.
- [x] I have manually tested this PR.

### CODE REVIEW

File: packages/web/components/Modal/BuyTechnologyModal/index.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+	{
+		id: 1,
+		label: 'Privado',
+		value: 'private',
+	},
+	{
+		id: 2,
+		label: 'Empresa',
+		value: 'company',
+	},
```
Comment (lcnogueira) [inline packages/web/components/Modal/BuyTechnologyModal/index.js:20]: Not sure if this something related to the code review itself, but aren't those options the same? I mean...private and company are both private... 🤔 

if it makes sense, we can remove one of them. Any thoughts, @JoaoPauloSMoura?

**Review by lcnogueira — CHANGES_REQUESTED** [review]
Well done, Felício 💯 

I think we can make some improvements to the mobile version in order to keep the visual pattern: 
- Add some margin-top to the title (it's too close to the thumbnail);
- Making the "Cancelar" and "Adquitir Tecnologia" buttons fill the available width;

File: packages/web/components/Modal/BuyTechnologyModal/index.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+				<div>
+					<h3>{technology.title}</h3>
+
+					<S.QuantityField>
+						<span id="btm1befa-technology-quantity">
+							Quantas unidades da tecnologia?
+							<RequiredIndicator />
+						</span>
+
+						<div aria-labelledby="btm1befa-technology-quantity">
```
Comment (lcnogueira) [inline packages/web/components/Modal/BuyTechnologyModal/index.js:114]: Looks great! 😍

File: packages/web/components/Modal/BuyTechnologyModal/index.js (hunk 1 above)
Comment (JoaoPauloSMoura) [reply packages/web/components/Modal/BuyTechnologyModal/index.js:20]: > Not sure if this something related to the code review itself, but aren't those options the same? I mean...private and company are both private... 🤔
> 
> if it makes sense, we can remove one of them. Any thoughts, @JoaoPauloSMoura?

Private is related to personal use, and company is for commercial use.

File: packages/web/components/Modal/BuyTechnologyModal/index.js (hunk 2 above)
Comment (luizeboli) [reply packages/web/components/Modal/BuyTechnologyModal/index.js:114]: Ty ❤️

**Review by calaca — COMMENTED** [review]
Great job @luizeboli ! It's working well! :clap: 

I do believe it's best to apply those changes suggested by @lcnogueira . They'll definitely improve mobile UI by a lot.

File: packages/web/components/Modal/BuyTechnologyModal/index.js (hunk 1 above)
Comment (luizeboli) [reply packages/web/components/Modal/BuyTechnologyModal/index.js:20]: Ty @JoaoPauloSMoura

**Comment by luizeboli** [comment]
Ty @lcnogueira and @calaca 

I've just improved UX on lower breakpoints, it's really better now 😃 :

image

**Comment by JoaoPauloSMoura** [comment]
> Ty @lcnogueira and @calaca
> 
> I've just improved UX on lower breakpoints, it's really better now 😃 :
> 
> image

Much better, you could switch buttons, "adquirir Tecnologia" first

**Comment by luizeboli** [comment]
Ty @JoaoPauloSMoura, done 🤝

File: packages/web/components/Technology/Details/Header.js [hunk 3]
```diff
@@ -1,12 +1,27 @@
...
 import { Likes, Share } from '../../Card';
 import ImagesCarousel from './ImagesCarousel';
 
 const Header = () => {
 	const { technology, implementationCosts } = useTechnology();
+	const { openModal } = useModal();
+	const { user } = useAuth();
+	const { t } = useTranslation(['common']);
+
+	const handleClick = ({ target: { name } }) => {
```
Comment (mateus4k) [inline packages/web/components/Technology/Details/Header.js:15]: Awesome trick for easier future feature expansion 🚀

File: packages/web/components/Modal/LoginModal/index.js [hunk 4]
```diff
@@ -31,7 +31,10 @@ const LoginModal = ({ message: incomingMessage, redirectTo }) => {
 			if (redirectTo) {
 				router.push(redirectTo);
 			}
+
 			closeModal();
+
+			onSuccessLogin();
```
Comment (mateus4k) [inline packages/web/components/Modal/LoginModal/index.js:37]: Unbelievable trick! 😱 🚀

**Review by mateus4k — APPROVED** [review]
All right! 🥇💯

**Review by calaca — APPROVED** [review]
Everything looks great! :clap:

File: packages/web/components/Modal/LoginModal/index.js (hunk 4 above)
Comment (luizeboli) [reply packages/web/components/Modal/LoginModal/index.js:37]: The man is really good

File: packages/web/components/Technology/Details/Header.js (hunk 3 above)
Comment (luizeboli) [reply packages/web/components/Technology/Details/Header.js:15]: Good man

File: packages/web/components/Modal/BuyTechnologyModal/index.js (hunk 1 above)
Comment (lcnogueira) [reply packages/web/components/Modal/BuyTechnologyModal/index.js:20]: Got it, @JoaoPauloSMoura . Makes sense 👍

**Review by lcnogueira — APPROVED** [review]
Great job, Felício. Well done! 💯

**Pull request merged**